scipy/stats/_sampling.py

```diff
@@ -0,0 +1,1234 @@
+        truncated, and the domain is inferred from the support of the
+        distribution.
+    loc : float, optional
+        The location parameter. Note that only floats and no arrays are
+        allowed. The default is 0.
+    scale : float, optional
+        The scale parameter. Note that an array of floatsis not allowed.
+        The default is 1.
+    ignore_shape_range : boolean, optional.
+        If False, shape parameters that are outside of the valid range
+        of values to ensure that the numerical accuracy (see Notes) is
+        high, raise a ValueError. If True, any shape parameters that are valid
+        for the distribution are accepted. This can be useful for testing.
+        The default is False.
+        If set to True, warnings are not suppresed that might occur when an
+        instance of the class is created.
+    random_state : {None, int, `numpy.random.Generator`,
+                        `numpy.random.RandomState`}, optional
+
+            A NumPy random number generator or seed for the underlying NumPy
+            random number generator used to generate the stream of uniform
+            random numbers.
+            If `random_state` is None, it uses ``self.random_state``.
+            If `random_state` is an int,
+            ``np.random.default_rng(random_state)`` is used.
+            If `random_state` is already a ``Generator`` or ``RandomState``
+            instance then that instance is used.
+
+    Attributes
+    ----------
+    loc : float
+        The location parameter.
+    random_state : {`numpy.random.Generator`, `numpy.random.RandomState`}
+        The random state used in relevant methods like `rvs` (unless
+        another `random_state` is passed as an argument to these methods).
+    scale : float
+        The scale parameter.
+
+    Methods
+    -------
+    cdf
+    evaluate_error
+    isf
+    logpdf
+    pdf
+    ppf
+    ppf_fast
+    qrvs
+    rvs
+    support
+    sf
+
+    Notes
+    -----
+    The class creates an object for continuous distributions specified
+    by `distname` with corresponding shape parameters args similar to a
+    frozen instance of a distribution. The method `rvs` uses a generator from
+    `scipy.stats.sampling` that is created when the object is instantiated.
+    In addition, the methods `qrvs` and `ppf_fast` are added.
+    `qrvs` generate samples based on quasi-random numbers from
+    `scipy.stats.qmc`. `ppf_fast` is the PPF based on the
+    numerical inversion method in [1]_ (`NumericalInversePolynomial`) that is
+    used to generate random variates.
+
+    Supported distributions (`distname`) are:
+    alpha, anglit, argus, beta, betaprime, bradford,
+    burr, burr12, cauchy, chi, chi2, cosine,
+    crystalball, expon, gamma, gennorm, geninvgauss,
+    gumbel_l, gumbel_r, hypsecant, invgamma, invgauss,
+    invweibull, laplace, logistic, maxwell, moyal,
+    norm, pareto, powerlaw, t, rayleigh, semicircular,
+    wald, weibull_max, weibull_min.
```

Comment:
It would be good to typeset these as either math (double backticks) or links (single backticks). That's anyway helpful, and also the underscores in the weibull distributions are special reST symbols.

Reply:
Ok, I will use double backticks (to make links work, I think I would need to write `scipy.stats.distname` since this file is in `scipy.stats.sampling`, which would be harder to read)